astropy/units/typing.py

```diff
@@ -0,0 +1,66 @@
+"""Typing module for supporting type annotations related to :mod:`~astropy.units`."""
+
+from __future__ import annotations
+
+__all__ = ["QuantityLike"]
+
+
+from typing import TYPE_CHECKING, Union
+
+import numpy.typing as npt
+
+from astropy.units import Quantity
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
```

Comment:
Since the next release of Astropy should be Python 3.10+, can we now do `from typing import TypeAlias`?

Reply:
Then `ruff` will fix the import for us.